ocw/lib/provider.py

```diff
@@ -50,3 +52,32 @@ def parse_image_name_helper(self, img_name, regex_s, group_key=['version', 'flav
                     'build': "-".join([gdict[k] for k in group_build if k in gdict and gdict[k] is not None]),
                 }
         return None
+
+    def get_keeping_image_names(self, images):
+        images_by_flavor = dict()
+        for img in images:
+            if (img.flavor not in images_by_flavor):
+                images_by_flavor[img.flavor] = list()
+            images_by_flavor[img.flavor].append(img)
+
+        keep_images = list()
+        for img_list in [images_by_flavor[x] for x in sorted(images_by_flavor)]:
+            img_list.sort(key=lambda x: LooseVersion(x.build), reverse=True)
+            for i in range(0, len(img_list)):
+                img = img_list[i]
+                if (not self.needs_to_delete_image(i, img.date)):
+                    keep_images.append(img.name)
+
+        return keep_images
+
+
+class Image:
+
+    def __init__(self, name, flavor, build, date):
+        self.name = name
+        self.flavor = flavor
+        self.build = build
```

Comment:
what build is this referred to? Shall we differentiate between PC build and kiwi build?

Reply:
It is the "Build" extracted from image name. Which is `<kiwi>-<PC Build>`. And we use LooseVersion() for parsing this version number and order. So I think not.